File: new-algorithm/src/javaalgo/programmers/가운데_글자_가져오기.java
package javaalgo.programmers;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class 가운데_글자_가져오기 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String s = br.readLine();

        System.out.println(solution(s));
    }

    static String solution(String s) {
        int mid = s.length() / 2;

        return s.length() % 2 == 0 ? s.substring(mid - 1, mid + 1) :s.substring(mid, mid + 1);
    }
}

File: new-algorithm/src/kotlinalgo/programmers/가운데_글자_가져오기.kt
package kotlinalgo.programmers

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val input = readLine()
    println(solution2(input))
}

fun solution2(s: String): String {

    return s.substring((s.length - 1) / 2, s.length / 2 + 1)

}